src/components/CriteriaFormV2/CriteriaForm.tsx

```diff
@@ -0,0 +1,371 @@
+        .string({
+            required_error: 'Title is required',
+        })
+        .min(1, {
+            message: 'Title must be longer than 1 symbol',
+        }),
+    selected: z
+        .object({
+            id: z.string(),
+            title: z.string(),
+        })
+        .nullish(),
+});
+
+type CriteriaFormValues = z.infer<typeof schema>;
+
+function patchZodSchema(data: ValidityData) {
+    const patched = schema.merge(
+        z.object({
+            title: schema.shape.title.refine((val) => !data.title.some((t) => t === val), {
+                message: 'Title must be unique',
+            }),
+            /* INFO: https://github.com/colinhacks/zod#abort-early */
+            weight: schema.shape.weight.superRefine((val, ctx): val is string => {
+                if (!val || !val.length) {
+                    return z.NEVER;
+                }
+
+                const parsed = Number(val);
+
+                if (Number.isNaN(parsed)) {
+                    ctx.addIssue({
+                        code: z.ZodIssueCode.custom,
+                        message: 'Weight must be an integer',
+                    });
+                }
+
+                if (parsed < minPossibleWeight || data.sumOfCriteria + parsed > maxPossibleWeight) {
+                    ctx.addIssue({
+                        code: z.ZodIssueCode.custom,
+                        message: 'Passed weight is not in range',
+                    });
+                }
+
+                return z.NEVER;
+            }),
+        }),
+    );
+
+    return patched;
+}
+
+interface CriteriaFormProps {
+    items: SuggestItem[];
+    defaultMode?: CriteriaFormMode;
+    withModeSwitch?: boolean;
+    values?: Partial<CriteriaFormValues>;
+    validityData: { title: string[]; sumOfCriteria: number };
+
+    onModeChange?: (mode: CriteriaFormMode) => void;
+    onSubmit: (values: CriteriaFormValues) => void;
+    onCancel: () => void;
+    onItemChange?: (item?: SuggestItem) => void;
+    onInputChange?: (value?: string) => void;
+
+    renderItem: React.ComponentProps<typeof AutoComplete<SuggestItem>>['renderItem'];
+}
+
+const radios: Array<{ value: CriteriaFormMode; title: string }> = [
+    { title: 'Simple', value: 'simple' },
+    { title: 'Goal', value: 'goal' },
+];
+
+const keyGetter = (item: SuggestItem) => item.id;
+
+const StyledFormInput = styled(FormInput)`
+    font-size: 14px;
+    font-weight: normal;
+    padding: 5px 10px;
+    width: 3ch;
+
+    border: 1px solid ${gray7};
+    box-sizing: content-box;
+`;
+
+interface WeightFieldProps {
+    name: 'weight';
+    value?: string;
+    onChange: ReactEventHandler<HTMLInputElement>;
+    error?: { message?: string };
+    maxValue: number;
+}
+
+const StyledWeightFieldWrapper = styled.div`
+    display: inline-flex;
+    align-items: center;
+    justify-content: flex-start;
+    gap: ${gapS};
+    height: 28px; // input height
+`;
+
+const StyledText = styled(Text)`
+    white-space: nowrap;
```

Comment:
Text has prop for that

Reply:
No, haven't
<img width="600" alt="image" src="https://github.com/taskany-inc/issues/assets/7002692/d544388d-e274-43ac-9289-a6d88bc44424">